### PR TITLE
Add tensorboard logging during training

### PR DESCRIPTION
**Pull request by justinchiu**
Embeddings don't load on tensorboard for me, so that is not included in this PR.

The pytorch 0.4 detach support can be taken it out, it was so I could get this running.

### CODE REVIEW

**Comment by justinchiu** [comment]
I'll separate the changes for pytorch 0.4 compat.

**Comment by srush** [comment]
Failing lint check. 

2.52s$ flake8
./train.py:14:1: F401 'torch.autograd.Variable as V' imported but unused
./train.py:14:1: N814 camelcase 'V' imported as constant
./train.py:78:1: E302 expected 2 blank lines, found 1
./train.py:104:1: E302 expected 2 blank lines, found 1
./onmt/Models.py:584:80: E501 line too long (99 > 79 characters)

**Comment by srush** [comment]
Awesome. Merging.
